classes/form/preferences_form.php

```diff
@@ -38,6 +38,11 @@ public function definition() {
                                  get_string('preference_mail_digest_help', 'local_extension'));
         $this->_form->setDefault('mail_digest', (new preferences())->get(preferences::MAIL_DIGEST));
 
+        $this->_form->addElement('checkbox', preferences::EXPORT_CSV,
```

Comment:
You can add `$preferences = new preferences();` at the beggining of the function and replace both `(new preferences())->get(...)` with `$preferences->get(...)` , basically you create one "preferences" object and use it twice instead of creating two objects and using once.